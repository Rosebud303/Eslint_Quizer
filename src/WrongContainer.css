.wrong-container {
    position: absolute;
    left: 50%;
    -webkit-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    background: #9E9E9E;
    padding: 30px;
    border: ridge 15px red;
    border-radius: 25px;
}