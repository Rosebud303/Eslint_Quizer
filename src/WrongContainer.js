import React, { Component } from 'react';
import WrongItem from './WrongItem.js';
import './WrongContainer.css';

export default class WrongContainer extends Component {
  constructor(props) {
    super(props);

  }

  
  render() {
    const { questions, incorrects, toggle } = this.props
    const getIncorrectObj = questions.filter(question => (incorrects.includes(question.id)));
    const displayWrong = getIncorrectObj.map(question => {
      return <WrongItem {...question} />
    })
    return (
      <div className='wrong-container'>
        <h2>Missed items...</h2>
        {displayWrong}
        <button className='show-wrong-button' onClick={toggle}>Finished</button>
      </div>
    )
  }
}
