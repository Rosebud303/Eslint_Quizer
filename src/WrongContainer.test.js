import React from 'react';
import WrongContainer from './WrongContainer.js';
import { shallow } from 'enzyme';

const mockQuests = [
    {
        "name": "Indentation",
        "quiz": "https://i.imgur.com/Mglgl9Y.png",
        "answer": "https://i.imgur.com/Tajm3yU.png",
        "id": 1
    },
    {
        "name": "Indentation",
        "quiz": "https://i.imgur.com/0pEnCe4.png",
        "answer": "https://i.imgur.com/pJN3TaQ.png",
        "id": 2
    },
    {
        "name": "No Unused Vars",
        "quiz": "https://i.imgur.com/7c32ZLd.png",
        "answer": "https://i.imgur.com/789uhJG.png",
        "id": 3
    },
    {
        "name": "Key Spacing",
        "quiz": "https://i.imgur.com/6F1XUXC.png",
        "answer": "https://i.imgur.com/gcQ23xw.png",
        "id": 4
    }
];

const mockIncorrect = [2,4];
const mockFunc = jest.fn();

describe('WrongContainer', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow (
            <WrongContainer questions={mockQuests}
                            incorrects={mockIncorrect}
                            toggle={mockFunc}
            />
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should simulate a click and call the function inside', () => {
        wrapper.find('.show-wrong-button').simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    });
});
