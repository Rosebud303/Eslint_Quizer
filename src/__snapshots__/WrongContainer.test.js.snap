// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrongContainer Should match the snapshot 1`] = `
<div
  className="wrong-container"
>
  <h2>
    Missed items...
  </h2>
  <WrongItem
    answer="https://i.imgur.com/pJN3TaQ.png"
    id={2}
    name="Indentation"
    quiz="https://i.imgur.com/0pEnCe4.png"
  />
  <WrongItem
    answer="https://i.imgur.com/gcQ23xw.png"
    id={4}
    name="Key Spacing"
    quiz="https://i.imgur.com/6F1XUXC.png"
  />
  <button
    className="show-wrong-button"
    onClick={[MockFunction]}
  >
    Finished
  </button>
</div>
`;
